docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -153,7 +157,921 @@ Future releases
+* This label signals that the pod has SPIRE integration enabled.
+* It prevents Kuma from delivering Spire-related configuration (e.g., SDS config, identity provider references) to pods that don't have the socket mounted.
+
+1. We could expose the configuration in control-plane to be more generic
+
+```yaml
+runtime:
+  kubernetes:
+    injector:
+        extraLabels:
+          kuma.io/spire-integration: enabled
+        extraVolumes:
+          - name: spire-agent-socket
+            hostPath:
+              path: /run/spire/sockets
+              type: Directory
+        containerPatches: ["spire-agent-socket-patch"]
+```
+
+ContainerPatch
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: ContainerPatch
+metadata:
+  name: spire-agent-socket-patch
+  namespace: kuma-system
+spec:
+  sidecarPatch:
+    - op: add
+      path: /volumeMounts/-
+      value:
+        name: spire-agent-socket
+        mountPath: /run/spire/sockets/
+        readOnly: true
+```
+
+3. We could introduce templating/overlay (out-of-scope for this MADR)
+
+We could introduce overlay to the control-plane injector config.
+
+Example: 
+
+```yaml
+runtime:
+  kubernetes:
+    injector:
+        overlay:
+            - path: spec.template.spec.volumes.spire-agent-socket
+              op: add
+              value:
+                name: spire-agent-socket
+                hostPath:
+                  path: /run/spire/sockets
+                  type: Directory
+            - path: spec.template.spec.containers.kuma-sidecar.volumeMounts.spire-agent-socket
+              op: add
+              value:
+                name: spire-agent-socket
+                mountPath: /run/spire/sockets/
+                readOnly: true          
+```
+
+#### Define active identity provider
+
+Previously, defining the active identity for a group of services was done through a single, perhaps less flexible, mechanism. Now, the `MeshIdentity` resource, leveraging its `selector`, offers a more granular and controlled approach to activate specific configurations.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshIdentity
+metadata:
+  name: old-identity
+  namespace: kuma-system # only in system namespace
+  labels:
+    kuma.io/mesh: default
+spec:
+  selector:
+    dataplane:
+      matchLabels: {} # select all
+....
+---
+apiVersion: kuma.io/v1alpha1
+kind: MeshIdentity
+metadata:
+  name: new-identity
+  namespace: kuma-system # only in system namespace
+  labels:
+    kuma.io/mesh: default
+spec:
+  selector:
+    dataplane:
+      matchLabels:
+        app: demo-app
+```
+
+The `MeshIdentity` resource’s selector field allows selection based on dataplane labels. Users can define it to apply to either all dataplanes or a specific subset.
+If there are multiple (X) identities selecting the same dataplane, the control plane will resolve the conflict by choosing one based on the following order of precedence:
+
+1. Selector specificity – the identity with most specific selector is preferred.
```

Comment:
This should be more rigourous on what specific mean. I think it's by number of keys in the selector.